packages/strapi-connector-bookshelf/lib/buildQuery.js

```diff
@@ -84,38 +84,52 @@ const buildJoinsAndFilter = (qb, model, whereClauses) => {
 
       qb.leftJoin(
         `${originInfo.model.databaseName}.${assoc.tableCollectionName} AS ${joinTableAlias}`,
-        `${joinTableAlias}.${singular(
-          destinationInfo.model.attributes[assoc.via].attribute
-        )}_${destinationInfo.model.attributes[assoc.via].column}`,
+        `${joinTableAlias}.${singular(destinationInfo.model.attributes[assoc.via].attribute)}_${
+          destinationInfo.model.attributes[assoc.via].column
+        }`,
         `${originInfo.alias}.${originInfo.model.primaryKey}`
       );
 
       qb.leftJoin(
         `${destinationInfo.model.databaseName}.${destinationInfo.model.collectionName} AS ${destinationInfo.alias}`,
-        `${joinTableAlias}.${singular(
-          originInfo.model.attributes[assoc.alias].attribute
-        )}_${originInfo.model.attributes[assoc.alias].column}`,
+        `${joinTableAlias}.${singular(originInfo.model.attributes[assoc.alias].attribute)}_${
+          originInfo.model.attributes[assoc.alias].column
+        }`,
         `${destinationInfo.alias}.${destinationInfo.model.primaryKey}`
       );
-      return;
-    }
+    } else if (assoc.nature === 'manyWay') {
+      const joinTableAlias = generateAlias(assoc.tableCollectionName);
 
-    const externalKey =
-      assoc.type === 'collection'
-        ? `${destinationInfo.alias}.${assoc.via ||
-            destinationInfo.model.primaryKey}`
-        : `${destinationInfo.alias}.${destinationInfo.model.primaryKey}`;
-
-    const internalKey =
-      assoc.type === 'collection'
-        ? `${originInfo.alias}.${originInfo.model.primaryKey}`
-        : `${originInfo.alias}.${assoc.alias}`;
-
-    qb.leftJoin(
-      `${destinationInfo.model.databaseName}.${destinationInfo.model.collectionName} AS ${destinationInfo.alias}`,
-      externalKey,
-      internalKey
-    );
+      qb.leftJoin(
+        `${originInfo.model.databaseName}.${assoc.tableCollectionName} AS ${joinTableAlias}`,
+        `${joinTableAlias}.${singular(originInfo.alias)}_id`,
```

Comment:
Two comments here:

- Test with a manyWay pointing to the same model ;) 
- As it is the only change can you refactor the code a bit to avoid repetition :)

Reply:
Ah oui bien vu le `related` 😏